feat(deletions): Request Seer deletions in batches

armenzg opened the pull request:
Some deletions are failing because the query to get group hashes times out.

Related: #inc-1236

Fixes [SENTRY-43X8](https://sentry.sentry.io/issues/6712841606/)
Fixes [SENTRY-3DVG](https://sentry.sentry.io/issues/5737861849/)
Fixes [SENTRY-3XEE](https://sentry.sentry.io/issues/6635786687/)

armenzg committed:
feat(deletions): Request Seer deletions in batches

Some deletions are failing because the query to get group hashes times out.

armenzg committed:
Fix

armenzg committed:
Remove this code

armenzg committed:
Schedule a task of BATCH_SIZE

JoshFerge commented on src/sentry/tasks/delete_seer_grouping_records.py:
```diff
@@ -14,6 +14,9 @@
 from sentry.taskworker.namespaces import seer_tasks
+from sentry.utils.query import RangeQuerySetWrapper
+
+BATCH_SIZE = 10000
```
how big will 10000 be in KB?

armenzg replied:
I've reduced this locally.

JoshFerge commented on src/sentry/tasks/delete_seer_grouping_records.py:
```diff
@@ -67,15 +70,20 @@ def call_delete_seer_grouping_records_by_hash(
-        )
+        group_hashes = []
+
+        for group_hash in RangeQuerySetWrapper(
```
can we add a test to make sure the right number of tasks gets called?

armenzg committed:
Add tests

armenzg committed:
Remove unneeded test

armenzg committed:
Fix test

armenzg committed:
tests(seer): Fix test for chunks

The `test_call_delete_seer_grouping_records_by_hash_chunked` test incorrectly uses `mock_batch_size`, which is a `MagicMock` object, in arithmetic operations and assertions. This causes a `TypeError` when calculating `range(mock_batch_size + 5)` and leads to incorrect comparisons (e.g., `len(...) == mock_batch_size`). The test should use the literal integer value `10` instead of the mock object for these operations.

This fixes it.

armenzg committed:
Fix test

lobsterkatie reviewed (approved):
LGTM!
